misk-redis/src/main/kotlin/misk/redis/RealRedis.kt

```diff
@@ -423,6 +431,129 @@ class RealRedis(
     )
   }
 
+  override fun zrange(
+    key: String,
+    type: ZRangeType,
+    start: ZRangeMarker,
+    stop: ZRangeMarker,
+    reverse: Boolean,
+    limit: ZRangeLimit?,
+  ): List<ByteString?> {
+    return zrangeBase(key, type, start, stop, reverse, false, limit)
+      .leftOrNull()?.map { it?.toByteString() } ?: listOf()
+  }
+
+  override fun zrangeWithScores(
+    key: String,
+    type: ZRangeType,
+    start: ZRangeMarker,
+    stop: ZRangeMarker,
+    reverse: Boolean,
+    limit: ZRangeLimit?,
+  ): List<Pair<ByteString?, Double>> {
+    return zrangeBase(key, type, start, stop, reverse, true, limit)
+      .getOrNull()?.map { Pair(it.binaryElement?.toByteString(), it.score) } ?: listOf()
+  }
+
+  private fun zrangeBase(
+    key: String,
+    type: ZRangeType,
+    start: ZRangeMarker,
+    stop: ZRangeMarker,
+    reverse: Boolean,
+    withScore: Boolean,
+    limit: ZRangeLimit?,
+  ): Either<List<ByteArray?>, List<Tuple>> {
+    return when (type) {
+      ZRangeType.INDEX ->
+        zrangeByIndex(key, start as ZRangeIndexMarker, stop as ZRangeIndexMarker, reverse,
+                      withScore)
+
+      ZRangeType.SCORE ->
+        zrangeByScore(key, start as ZRangeScoreMarker, stop as ZRangeScoreMarker, reverse,
+                      withScore, limit)
+
+      ZRangeType.LEX ->
+        throw RuntimeException("Unsupported UnifiedJedis implementation of BYLEX option for " +
+                                 "zrange command")
+    }
+  }
+
+  private fun zrangeByScore(
+    key: String,
+    start: ZRangeScoreMarker,
+    stop: ZRangeScoreMarker,
+    reverse: Boolean,
+    withScore: Boolean,
+    limit: ZRangeLimit?,
+  ): Either<List<ByteArray?>, List<Tuple>> {
+    val minString = start.toString()
+    val maxString = stop.toString()
+
+    return if (limit == null && !reverse && !withScore) {
+      Either.Left(unifiedJedis.zrangeByScore(key.toByteArray(charset),
```

Comment:
This indent formatting doesn't look idiomatic to Kotlin, also would prefer Tuple/Pair than introducing Either.

Reply:
will remove.